### ISSUE TITLE
Use `GlobalConfiguration` not `config-file-provider`

### ISSUE THREAD

**Issue by jglick** [samrocketman/restrict-steps-plugin#3]
In fact you ought to be able to provide GUI configuration that offers completion on `StepDescriptor`s.

**Comment by samrocketman** [comment]
This was just POC so I prioritized getting something working.  Totally open to using a UI config and removing the config file provider plugin dependency.

It wasn't clear that it would actually work initially.  But now it does so sure can switch to a more Jenkins-friendly config

**Comment by jglick** [comment]
> using a UI config

(or CasC)

**Comment by samrocketman** [comment]
Currently working on hosting.  I'd like to move this issue to Jenkins jira once the hosting request is complete.

**Comment by samrocketman** [comment]
I have migrated this to https://issues.jenkins.io/browse/JENKINS-71574; please use your Jira powers to update the reporter to yourself and feel free to edit the description.

I will leave this message up until next week which is when I will delete this repository so that the hosted plugin unlinks its fork.

**Comment by jglick** [comment]
> please use your Jira powers to update the reporter to yourself

I lack that power.

**Comment by samrocketman** [comment]
> > please use your Jira powers to update the reporter to yourself
> 
> I lack that power.

ok if you want author tracking file a new ticket and I'll close the one I opened

**Comment by jglick** [comment]
No need.
